dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathResult.java

```diff
@@ -69,4 +71,24 @@ public ClassPathResult(
   ImmutableList<UnresolvableArtifactProblem> getArtifactProblems() {
     return artifactProblems;
   }
+
+  /** Returns text describing dependency paths to {@code jars} in the dependency tree. */
+  public String formatDependencyPaths(Iterable<Path> jars) {
+    StringBuilder message = new StringBuilder();
+    for (Path jar : jars) {
```

Comment:
This could be simpler if this method only formatted a single jar. Then you'd only need one loop in LinkageChecker.main and you wouldn't have to build an intermediate list.

Reply:
The intermediate ImmutableSet is for deduplication logic.